viadot/utils.py

```diff
@@ -483,4 +484,153 @@ def wrapper(*args, **kwargs) -> pd.DataFrame:
 
 
 def get_fqn(table_name: str, schema_name: str = None) -> str:
-    return f"{schema_name}.{table_name}" if schema_name else table_name
\ No newline at end of file
+    return f"{schema_name}.{table_name}" if schema_name else table_name
+
+
+def validations(
+    df: pd.DataFrame,
+    tests: dict = None,
+    stream_level: int = logging.INFO,
+    logger: Optional[logging.Logger] = None,
+) -> None:
+    """
+    Task to validate the data on DataFrame level. All numbers in the ranges are inclusive.
+    tests:
+        - `column_size`: dict{column: size}
+        - `column_unique_values`: list[columns]
+        - `column_list_to_match`: list[columns]
+        - `dataset_row_count`: dict: {'min': number, 'max', number}
+        - `column_match_regex`: dict: {column: 'regex'}
+        - `column_sum`: dict: {column: {'min': number, 'max': number}}
+    Args:
+        df (pd.DataFrame): The data frame for validation.
+        tests (dict, optional): Tests to apply on the data frame. Defaults to None.
+    Raises:
+        ValidationError: If validation failed for at least one test.
+    """
+
+    if logger is None:
+        logging.basicConfig(level=logging.INFO)
+        logger = logging.getLogger("prefect_shell.utils")
+
+    failed_tests = 0
+    failed_tests_list = []
+
+    if tests is not None:
+        if "column_size" in tests:
```

Comment:
I think each test should be split into a separate function and here you should just be calling the correct functions. This function is already too long and it will only grow. Also, the way this is done here, there's no way to unit test specific tests.